pkg/types/dsse/dsse.go

```diff
@@ -18,67 +18,76 @@ import (
 	"bytes"
 	"context"
 	"crypto"
-	"encoding/json"
+	"crypto/sha256"
+	"encoding/base64"
 	"errors"
 	"fmt"
 
+	v1 "github.com/sigstore/protobuf-specs/gen/pb-go/common/v1"
+
 	"github.com/secure-systems-lab/go-securesystemslib/dsse"
+	pbdsse "github.com/sigstore/protobuf-specs/gen/pb-go/dsse"
 	pb "github.com/sigstore/rekor-tiles/pkg/generated/protobuf"
 	"github.com/sigstore/rekor-tiles/pkg/pki/x509"
 	"github.com/sigstore/rekor-tiles/pkg/types/verifier"
 	"github.com/sigstore/sigstore/pkg/signature"
 	sigdsse "github.com/sigstore/sigstore/pkg/signature/dsse"
 )
 
-func Validate(ds *pb.DSSERequest) error {
-	if ds.Envelope == "" {
-		return fmt.Errorf("missing envelope")
+func ToLogEntry(ds *pb.DSSERequestV0_0_2) (*pb.DSSELogEntryV0_0_2, error) {
```

Comment:
I think this should be split into two functions, one for proto validity, and one for cryptographic validity, or we need a name that conveys that more clearly.

Reply:
Sure? There's a lot of work in validate that is useful to conversion. I could just make the conversion function always call validate before. But logically it seems like we'd be unlikely to call validate and tologentry separately

Reply:
I’m not super opinionated on if we should refactor into smaller functions, but this name isn’t conveying that this function does a) validation of request input, b) signature verification, c)converting request to entry. I think we could have each of those as unexported functions (which we can export later if they’d be useful in a library) and then one public function that calls all three.

Reply:
I also like this design because we can decide later to add back something like the Entry interface (which is overkill now since we have just two types)

Reply:
updated with comment, will followup in #178